hack/test-all.sh

```diff
@@ -3,6 +3,9 @@
 set -e -x -u
 
 ./hack/build.sh
+
+export KAPP_BINARY_PATH="$PWD/kapp"
```

Comment:
probably better to put that into ./hack/test-e2e.sh instead.

Reply:
I considered it, but we don't actually build the binary in `./hack/test-e2e.sh` so there was no guarantee that it was going to be in that location. Therefore, I thought it was better to default it the one that is on your path. I will move it there if you think it's better.

Reply:
@jfmyers9 i think it's better there, since a lot of times i found myself running individual tests, for example, `./hack/test-e2e.sh -run TestTemplate`. they still should use locally built binary.

Reply:
I added something to address this in `test-e2e.sh`.